plutus-tx-plugin/test/Plugin/NoTrace/WithTraces.hs

```diff
@@ -0,0 +1,40 @@
+{-# LANGUAGE BlockArguments    #-}
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell   #-}
+{-# LANGUAGE TypeApplications  #-}
+{-# OPTIONS_GHC -fplugin PlutusTx.Plugin #-}
+-- The plugin uses non-conservative optimizations by default and they remove some traces.
+-- We disable them to make sure that no traces are removed.
+{-# OPTIONS_GHC -fplugin-opt PlutusTx.Plugin:conservative-optimisation #-}
```

Comment:
the plugin options take effect in the module where you call `compile`, so if you wanted you could share the code between the two modules here, but not a big deal